### ISSUE TITLE
Schematic Clarification Please?

### ISSUE THREAD

**Issue by owlyowl** [decodeais/Raspi_onoff#1]
![image](https://github.com/user-attachments/assets/f630eacd-d8b8-457e-a2d9-d9a5c7eda3aa)

Hi I'm trying to get some clarification on a few things please:
1. What type of switch are you using there?
2. Are the two 3.3v sources coming from the Pi GPIO or a separate source?
3. What is the symbol with the two arrows as it could be an earth or a diode of sorts?
4. What role do the resistors play in the 3.3v feeds please and why are there two on the one GPIO pin?

**Comment by decodeais** [comment]
D1 is the power led red green , how you like
The normal push button: 
	closed if you press it
	open when you do not
The values of the resistors are not very important. Double or half should work too.
Of cause I use the 3.3v from the header.
R1 was the pull up. I left it to show the value if you don' use the optional power led. 
R2 is to limit the Basis current if you put the gpio27 to high level by accident.  If you are Shure that you never do it you can connect it directly.
For the led you need a lower value.
The dammed schematic app had only this sign for ground.

**Comment by owlyowl** [comment]
D1 is the raspberry Pi's power LED or a separate one please?

So a latching push button like so?
https://au.mouser.com/ProductDetail/E-Switch/TL2285EE?qs=7Bi7Dg8SEnxrzf04kzdLHA%3D%3D&mgh=1&vip=1&utm_id=17753636782&gad_source=1&gclid=CjwKCAjw_4S3BhAAEiwA_64Yhp31UdLwBvHMZsaU8Y8-kn1yJJhVytmc03WfOr9H4LwW2G9b10gjPBoCAgAQAvD_BwE

And then where does the D1 led ground itself sorry? Sorry for the newbie questions

**Comment by decodeais** [comment]
The negativ pin is connected the gpio27. 
The switch is  an double switchover. It is perfekt for the solution on the i2c bus pins. You need only the cheapest version: on contact closed when pressed. You can use the changeover switch to, but then it's a little more difficult to find the to pins which are closed when you press it

**Comment by decodeais** [comment]
Don' t forget to order the knob for the button. Then it looks much better. It has always to be ordered separately. My englisch so god so it last sometimes to check the vocabulary.
Hint for newbies: the led has one longer wire, this is plus.
When you already cut it you can see the bottom is not complete roun. The flat edge is minus.

**Comment by owlyowl** [comment]
Do you have any examples of components for the button/led etc?
[RaspberryPi-StartUpShutdown.zip](https://github.com/user-attachments/files/16963029/RaspberryPi-StartUpShutdown.zip)

I started a Kicad with both options you outlined, one with LED and one without.

I was wondering if you can use another pin other than 27 and 17 because I'm using them for a screen..

**Comment by decodeais** [comment]
You can use every pin. But don't forget to change the number in the initialisation. The cheapest way is to buy a kit for arduino[link to example](https://www.ebay.de/itm/174942171074?mkcid=16&mkevt=1&mkrid=707-127634-2357-0&ssspo=p795EKDWQse&sssrc=4429486&ssuid=NEHSZeXMRk6&var=&widget_ver=artemis&media=COPY)
I think you find something similar in the uk.

**Comment by owlyowl** [comment]
Sorry is the switch type
1. Momentary?
2. Latch switch please? - Push and stays in.. push again, springs out?
or like this:

**Comment by decodeais** [comment]
It should close only while you press. For me a push button always close or open directly. The version that stays in the last position would be a switch. A push button has no memory Funktion inside, but I am no native speaker.

**Comment by owlyowl** [comment]
> It should close only while you press. For me a push button always close or open directly. The version that stays in the last position would be a switch. A push button has no memory Funktion inside, but I am no native speaker.

Like an arcade machine button?

**Comment by decodeais** [comment]
Like a door bell push button.

Tom Fulcher ***@***.***> schrieb am Do., 12. Sep. 2024, 12:24:

**Comment by owlyowl** [comment]
Perfect thank you.. thanks for your patience.

Do you have a photo of your setup?

**Comment by decodeais** [comment]
I tested a little change in the schematic. 
You do not have to use the run test pad. It works with the gpio3 too.  I wanted to be Shure that I have a working i2c bus til the end, because  I wanted to have access to the display in the shutdown Sequenz to clear the display.  If you are interested in my little project you can look to  my  [WebRadio repository ](https://github.com/decodeais/WebRadio) .

**Comment by owlyowl** [comment]
![image](https://github.com/user-attachments/assets/6011d108-b953-490e-988a-d470436d7867)
![PXL_20240924_111947867 MP](https://github.com/user-attachments/assets/20a00829-4ae7-4f86-ae9d-3bfbec63487d)
![PXL_20240924_111715100 MP](https://github.com/user-attachments/assets/25166178-392c-4308-be7b-ae18023d7596)
![PXL_20240924_111710165 MP](https://github.com/user-attachments/assets/2a1bd58d-acb0-4555-8e8c-c927a3ade84c)

I have wired this up and tried but the LED never turns on and shutdown never initiates.

**Comment by decodeais** [comment]
I'm very ill in this moment but I will try to help you. You should test
shutdown and wake up separately. First you should test shutdown. If you
install only the wires from the button to the gpio pin for shutdown and
initialize the pin for shutdown in the config.txt.

Tom Fulcher ***@***.***> schrieb am Di., 24. Sep. 2024, 13:22:

**Comment by owlyowl** [comment]
If I just have GPIO 24 and a button with no GPIO 23 plugged into the bread board it shuts down. Run also needs to be off the breadboard too.

Please don't worry if you're ill.. health and people come first :)

**Comment by decodeais** [comment]
If you have initialized  the shutdown pin. It is enough to touch it with a wire to ground.
If it does not work there is a typo in the initialisation ore a mismatch with the pins. Did you check the switch with a multimeter?

**Comment by owlyowl** [comment]
Yes GPIO24 will shut down perfectly bringing it to ground with just it and the momentary push button.

If I don't involve having the RUN pin plugged in or the BC337.

**Comment by decodeais** [comment]
If the the raspi runs. The voltage at the gpio23 has to be zero to switch of the transistor. Has this output zero volts?
Why did you add D1 ?

**Comment by owlyowl** [comment]
I removed D1 but yes still no change.. I'll grab my multimeter and test gpio 23

**Comment by decodeais** [comment]
I looked to your breadboard. I think you have placed the transistor wrong. It looks like you connected your pins by your breadboard. Turn the transistor 90° and start connecting the wires.

**Comment by owlyowl** [comment]
OMG I'm so sorry yes lack of sleep! Sure did!

**Comment by owlyowl** [comment]
That works perfectly thanks so much!

**Comment by decodeais** [comment]
I did not like the solution of the double use of the bus pins. I searched a while, then I had this idea, may be it would have been a good solution for old raspi without the wake-up feature. I think I am a little late.

**Comment by owlyowl** [comment]
I really appreciate you taking the time :)
